### DISCUSSION TITLE
NetBox v3.4.1 - [API] Can't create specific IP addresses in prefix

### DISCUSSION THREAD

**Discussion by UnLuckyAki** [netbox-community/netbox#13613] [Q&A] [answered]
Hello everyone! Can u give me some klues?
I'm using API command to create specific addresses inside prefix when it's created:
```
POST /ipam/prefixes/{id}/available-ips/
``` 
For example here is my json:
```javascript
[
  {
    "family": 4,
    "address": "185.31.34.4/29",
    "vrf": null,
    "tenant":4
  },
  {
    "family": 4,
    "address": "185.31.34.5/29",
    "vrf": null,
    "tenant":4
  }
]
``` 
But in NetBox I get 185.31.34.1 and 185.31.34.2.

**Comment by Sader82** [comment]
Having the same issue here! Is there anyone with a solution?

via dev tools in the browser, I see this: 
![2023-12-01_08h12_04](https://github.com/netbox-community/netbox/assets/25772690/1c5af3b9-8684-4fd0-b2f8-f54502e085ed)

**Reply by candlerb** [reply]
@Sader82: your problem is probably different, so please start from the beginning, preferably in a new thread.

Since this is "form data" I guess you're talking about using the web interface not the API. So you need to give the full series of steps to reproduce the issue, for anyone to help you.

All I can see from the dev tools output is that you have a conflicting address in the query string and in the form data - but I have no idea how you got to that point.

You also didn't say what Netbox version you're using.

**Comment by candlerb** [comment, marked as answer]
@UnLuckyAki: The information you give is incomplete and inconsistent.

1. You say you are POSTing to `/ipam/prefixes/{id}/available-ips/`, but you say you are using the API. If that's true, it would be `POST /api/ipam/prefixes/{id}/available-ips/`
2. What specific prefix in Netbox are you referring to? Is it 185.31.34.0/29, or 185.31.34.8/29, or something else? What does `GET /api/ipam/prefixes/{id}/` show?
3. You say you are using POST, not GET.  However the JSON you show is what you'd get as the response to a GET request.  (Also, if it were a POST request, normally you'd need to show both the request body and the response body. In this case I believe the request body can be empty, although I think you still need to set `Content-Type: application/json`)
4. When you say "But in NetBox I get 185.31.34.1 and 185.31.34.2", what do you mean? Do you mean you did something in the Netbox web UI, if so what? What path do you see in the URL bar of your browser when you do this?
5. The title of this post is "Can't create specific IP addresses in prefix". If you know already which IP address you want to create, you should simply create that address by POSTing to `/api/ipam/ip-addresses/`. You would only use the `.../available-ips/` endpoints if you want Netbox to show you what's available (GET) or pick the first available automatically (POST) - in both cases you are *not* choosing a specific IP address. The difference between GET and POST is described [here](https://docs.netbox.dev/en/stable/release-notes/version-2.1/#automatic-provisioning-of-next-available-ip-1246).

**Comment by Sader82** [comment]
@candlerb Now I see what I did wrong.

I did a POST with /api/ipam/ip-addresses/ "10.10.10.1" and I forgot to add the prefix after the IP. So /api/ipam/ip-addresses/ "10.10.10.1/24" adds the IP address to the correct prefix.

**Reply by candlerb** [reply]
I think what you're talking about is the prefix *length*, i.e. the /24 part of the address.

In Netbox, the prefix length is an attribute of the IP address itself: for example 10.10.10.1/24. This is not like any other IPAM system I've ever seen, where an individual IP address would be just the address, 10.10.10.1
